### PR TITLE
Redesign of the product feed page

### PR DESCRIPTION
**Pull request by amaury-hanser**
# Redesign of the product feed page

## New component, not in use yet

image

There is a big todo for the front-end dev.

```
  // TODO
  /**
   * Create data structure for the sync history (vertical timeline)
   * An ENUM has been written in _dev/src/enums/product-feed/sync-history.ts
   * Use $t('productFeedPage.syncSummary.syncHistory.title.{$foo}') for the title
   * (with foo being planned | started | failed | completed | next)
   * Use $t('productFeedPage.syncSummary.syncHistory.subtitle.{$foo}', {date}) for the subtitle
   * (with foo being willHappenOnDate | happenedOnDate | error)
   * Icons and line colors are handled with css class:
   * Icons might be: sync-history__item--icon-{foo}
   * (with foo being check | schedule | cancel | change_circle)
   * Line colors might be: sync-history__item--bg-{foo}, the line is to applied on the top
   * (with foo being blue | red)
   */
```

### CODE REVIEW

File: _dev/src/components/product-feed-page/sync-overview.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    </template>
+    <div class="d-sm-flex">
+      <div class="flex-shrink-0 d-flex flex-column p-3">
+        <SyncHistory
+          v-slot="{ syncStates }"
+        >
+          <SyncState
+            v-for="syncState, i in syncStates"
+            :key="i"
+            v-bind="{...syncState}"
```
Comment (elisegriset92) [inline _dev/src/components/product-feed-page/sync-overview.vue:18]: not sure about this way of passing props as a destructured object, might need to check with @Quetzacoalt91  ?

**Comment by elisegriset92** [comment]
Should we create a new branch before merging to master ?

File: _dev/src/components/product-feed-page/sync-overview.vue (hunk 1 above)
Comment (amaury-hanser) [reply _dev/src/components/product-feed-page/sync-overview.vue:18]: Would you suggest it to be more explicit ? 

```
:title="syncState.title"
:description="syncState.description"
:icon="syncState.icon"
lineColor="syncState.lineColor"
```

Or to pass an object and change `syncState` component to expect only one props `syncState` ?

Comment (elisegriset92) [reply _dev/src/components/product-feed-page/sync-overview.vue:18]: No for me it's just 
`:syncState`
I don't understand why we would need to write `{...syncState}`

Comment (amaury-hanser) [reply _dev/src/components/product-feed-page/sync-overview.vue:18]: `v-bind="{...syncState}"` is equivalent to:
```
:title="syncState.title"
:description="syncState.description"
:icon="syncState.icon"
:lineColor="syncState.lineColor"
```

`:syncState="syncState"` is equivalent to:
```
:syncState="syncState"
```

In the first case, the destructuration allows to pass multiple props at once, 
In the second case, you pass only one props.

I went with the first solution to be able to add a validator for each props, but if it's not a best practice, I'm eager to change it.

Comment (amaury-hanser) [reply _dev/src/components/product-feed-page/sync-overview.vue:18]: You were correct before, I can do: `v-bind="syncState`.

About the validator, you're right, I should be able to validate the object.

**Pull request merged**